pkg/doc/verifiable/credential_sdjwt.go

```diff
@@ -0,0 +1,164 @@
+)
+
+type marshalDisclosureOpts struct {
+	includeAllDisclosures bool
+	discloseIfAvailable   []string
+	discloseRequired      []string
+	holderBinding         *holder.BindingInfo
+}
+
+// MarshalDisclosureOption provides an option for Credential.MarshalWithDisclosure.
+type MarshalDisclosureOption func(opts *marshalDisclosureOpts)
+
+// TODO: should DiscloseGiven(IfAvailable|Required) have path semantics for disclosure?
+
+// DiscloseGivenIfAvailable sets that the disclosures with the given claim names will be disclosed by
+// Credential.MarshalWithDisclosure.
+//
+// If any name provided does not have a matching disclosure, Credential.MarshalWithDisclosure will skip the name.
+//
+// Will result in an error if this option is provided alongside DiscloseAll.
+func DiscloseGivenIfAvailable(disclosureNames []string) MarshalDisclosureOption {
+	return func(opts *marshalDisclosureOpts) {
+		opts.discloseIfAvailable = disclosureNames
+	}
+}
+
+// DiscloseGivenRequired sets that the disclosures with the given claim names will be disclosed by
+// Credential.MarshalWithDisclosure.
+//
+// If any name provided does not have a matching disclosure, Credential.MarshalWithDisclosure will return an error.
+//
+// Will result in an error if this option is provided alongside DiscloseAll.
+func DiscloseGivenRequired(disclosureNames []string) MarshalDisclosureOption {
+	return func(opts *marshalDisclosureOpts) {
+		opts.discloseRequired = disclosureNames
+	}
+}
+
+// DiscloseAll sets that all disclosures in the given Credential will be disclosed by Credential.MarshalWithDisclosure.
+//
+// Will result in an error if this option is provided alongside DiscloseGivenIfAvailable or DiscloseGivenRequired.
+func DiscloseAll() MarshalDisclosureOption {
+	return func(opts *marshalDisclosureOpts) {
+		opts.includeAllDisclosures = true
+	}
+}
+
+// DisclosureHolderBinding option configures Credential.MarshalWithDisclosure to include a holder binding.
+func DisclosureHolderBinding(binding *holder.BindingInfo) MarshalDisclosureOption {
+	return func(opts *marshalDisclosureOpts) {
+		opts.holderBinding = binding
+	}
+}
+
+// MarshalWithDisclosure marshals a SD-JWT credential in combined format for presentation, including precisely
+// the disclosures indicated by provided options, and optionally a holder binding if given the requisite option.
+func (vc *Credential) MarshalWithDisclosure(opts ...MarshalDisclosureOption) (string, error) {
+	options := &marshalDisclosureOpts{}
+
+	for _, opt := range opts {
+		opt(options)
+	}
+
+	if options.includeAllDisclosures && (len(options.discloseIfAvailable) > 0 || len(options.discloseRequired) > 0) {
+		return "", fmt.Errorf("incompatible options provided")
+	}
+
+	// TODO: should make JWTCredClaims and sign if vc.JWT not available?
+	//  right now this method assumes that the VC was parsed from an SD-JWT, so vc.JWT is set,
+	//  and vc.SDJWTDisclosures has the parsed DisclosureClaims
```

Comment:
Question for reviewers: should I extend this method (and add additional options) so it can turn a regular VC into a SD-JWT VC?

Reply:
sounds like a good idea

Reply:
Done